scripts/lib/libvirt/libvirt_setup.py

```diff
@@ -221,14 +229,26 @@ def net_start(args):
     netpath = args.netpath
     netname = xml_get_value(args.netpath, "name")
     print("defining network from {0}".format(netname))
-    networks = conn.listNetworks()
-    if netname not in networks:
-        print("defining network from {0}".format(netpath))
-        xml = readfile(netpath)
-        conn.networkDefineXML(xml)
-    print("starting {0} network".format(netname))
-    dom = conn.networkLookupByName(netname)
-    dom.create()
+
+    # Get the names of active and inactive network domains.
+    try:
```

Comment:
Basically we want to catch an exception when we know what to do with this. In the current code we are ignoring the exceptions and printing a text. This means that in reality we don;t know what to do with these exceptions. So the best policy is move the management of the exception in the upper layer.

In this library we have two upper layers: the `main()` function and the `mkcloud` library user.  In the `main()` function _we_ know what to do with the exception: `sys.exit()` with a nice error code (and print the full trace of the exception to understand what happened)